### PR TITLE
Add close() to Env, add functionality to call close() on exit

### PR DESCRIPTION
**Pull request by jietang**
First pass at adding close-on-exit functionality to envs (the doom tests leave horrible dangling processes otherwise)

### CODE REVIEW

File: gym/core.py [hunk 1]
```diff
@@ -155,6 +162,17 @@ def render(self, mode='human'):
 
         return self._render(mode=mode, close=close)
 
+    def close(self):
```
Comment (gdb) [inline gym/core.py]: This should probably be idempotent. Maybe maintain a `closed` variable?

File: gym/utils/atexit_utils.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import atexit
```
Comment (gdb) [inline gym/utils/atexit_utils.py]: (I assume the plan is to DRY this up with the monitor `atexit` stuff?)

Comment (jietang) [inline gym/utils/atexit_utils.py]: Yup.

File: gym/core.py (hunk 1 above)
Comment (jietang) [inline gym/core.py]: Good idea. I'll make the change.

File: gym/monitoring/monitor.py [hunk 3]
```diff
@@ -46,20 +39,8 @@ def capped_cubic_video_schedule(episode_id):
...
-
-@atexit.register
-def close_all_monitors():
-    # Explicitly fetch all monitors first so that they can't disappear while
-    # we iterate. cf. http://stackoverflow.com/a/12429620
-    open_monitors = list(monitors.items())
-    for key, monitor in open_monitors:
-        monitor.close()
+def get_running_monitors():
+    return list(monitor_close_registry.close_objects.values())
```
Comment (gdb) [inline gym/monitoring/monitor.py]: I don't love the name `monitor_close_registry` BTW — could we do something like `monitor_closer`?

File: gym/monitoring/__init__.py [hunk 4]
```diff
@@ -1,3 +1,3 @@
-from gym.monitoring.monitor import Monitor, load_results, monitors as _monitors
+from gym.monitoring.monitor import Monitor, load_results, get_running_monitors
```
Comment (gdb) [inline gym/monitoring/__init__.py]: The intention with `_monitors` is to make clear that this is for use within OpenAI Gym only, and not within user code. I would keep a `_...` name here.

Maybe better to do `get_running_monitors` -> `open_monitors`?

**Pull request closed without merging**